### ISSUE TITLE
py3: "cannot use a string pattern on a bytes-like object"

### ISSUE THREAD

**Issue by hawkrives** [JimmXinu/FanFicFare#339]
(This is the python3 version of fanficfare.)

```
fanficfare --debug --version
FFF: DEBUG: 2018-09-24 14:32:16,299: cli.py(184):     OS Version:Darwin-18.0.0-x86_64-i386-64bit
FFF: DEBUG: 2018-09-24 14:32:16,299: cli.py(185): Python Version:3.7.0 (default, Jun 29 2018, 20:13:13)
[Clang 9.1.0 (clang-902.0.39.2)]
FFF: DEBUG: 2018-09-24 14:32:16,300: cli.py(186):    FFF Version:3.0.0
Version: 3.0.0
```

---

```
$ fanficfare --progress -u Harry\ Potter\ and\ the\ Heartlands\ of\ Time\ -\ joe6991.epub
Updating Harry Potter and the Heartlands of Time - joe6991.epub, URL: b'https://www.fanfiction.net/s/6325846/1/'
Traceback (most recent call last):
  File "/usr/local/bin/fanficfare", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/site-packages/fanficfare/cli.py", line 290, in main
    passed_personalini)
  File "/usr/local/lib/python3.7/site-packages/fanficfare/cli.py", line 338, in do_download
    output_filename)
  File "/usr/local/lib/python3.7/site-packages/fanficfare/cli.py", line 515, in get_configuration
    configuration = Configuration(adapters.getConfigSectionsFor(url), options.format)
  File "/usr/local/lib/python3.7/site-packages/fanficfare/adapters/__init__.py", line 279, in getConfigSectionsFor
    (cls,fixedurl) = _get_class_for(url)
  File "/usr/local/lib/python3.7/site-packages/fanficfare/adapters/__init__.py", line 288, in _get_class_for
    fixedurl = re.sub(r"(?i)^[htp]+(s?)[:/]+",r"http\1://",url.strip())
  File "/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/lib/python3.7/re.py", line 192, in sub
    return _compile(pattern, flags).sub(repl, string, count)
TypeError: cannot use a string pattern on a bytes-like object
```

This story was downloaded … probably two years ago, so definitely with the Python 2 code. I don't know how `-u` extracts the URL to fetch, but it seems to be getting a bytestring instead of a unicode one.

Python 2-originating file: [Harry Potter and the Heartlands of Time - joe6991.epub.zip](https://github.com/JimmXinu/FanFicFare/files/2412560/Harry.Potter.and.the.Heartlands.of.Time.-.joe6991.epub.zip)

---

If I download the story through the python3 version, I receive the same error.

```
$ fanficfare --progress -u Harry\ Potter\ and\ the\ Heartlands\ of\ Time-ffnet_6325846--py3.epub
Updating Harry Potter and the Heartlands of Time-ffnet_6325846--py3.epub, URL: b'https://www.fanfiction.net/s/6325846/1/'
Traceback (most recent call last):
  File "/usr/local/bin/fanficfare", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/site-packages/fanficfare/cli.py", line 290, in main
    passed_personalini)
  File "/usr/local/lib/python3.7/site-packages/fanficfare/cli.py", line 338, in do_download
    output_filename)
  File "/usr/local/lib/python3.7/site-packages/fanficfare/cli.py", line 515, in get_configuration
    configuration = Configuration(adapters.getConfigSectionsFor(url), options.format)
  File "/usr/local/lib/python3.7/site-packages/fanficfare/adapters/__init__.py", line 279, in getConfigSectionsFor
    (cls,fixedurl) = _get_class_for(url)
  File "/usr/local/lib/python3.7/site-packages/fanficfare/adapters/__init__.py", line 288, in _get_class_for
    fixedurl = re.sub(r"(?i)^[htp]+(s?)[:/]+",r"http\1://",url.strip())
  File "/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/lib/python3.7/re.py", line 192, in sub
    return _compile(pattern, flags).sub(repl, string, count)
TypeError: cannot use a string pattern on a bytes-like object
```

Python 3-originating file: [Harry Potter and the Heartlands of Time-ffnet_6325846--py3.epub.zip](https://github.com/JimmXinu/FanFicFare/files/2412591/Harry.Potter.and.the.Heartlands.of.Time-ffnet_6325846--py3.epub.zip)

**Comment by JimmXinu** [comment]
Fix is in test version v3.0.3 and in the next release version.
```
pip install --extra-index-url https://testpypi.python.org/pypi --upgrade FanFicFare 
```

**Comment by hawkrives** [comment]
Great, thanks!
